Core/src/main.cpp

```diff
@@ -142,9 +140,8 @@ int run_glfw() {
                     fixedUpdateBuffer -= PHYSICS_UPDATE_INTERVAL;
                 }
                 std::this_thread::sleep_for(std::chrono::microseconds(1));
-                audioManager->updateListenerPosition(cameraEnt->transform->getWorldPosition());
-                audioManager->updateSoundPosition(testCarEntity->transform->getWorldPosition());
-                audioManager->Update(0.0);
+                audioManager->updateListenerPosition(cameraEnt);
```

Comment:
This should be done interally in the system. This thread should just be calling audio manager's update.

Reply:
Notes: AudioManager doesn't have a parent entity with a transform. Realistically, this should actually be done by the developer, rather than the engine, as a developer may want to move the listener irrespective of an entity. Any thoughts on how to deal with this?

Reply:
If we go by Unity's spec, each scene should have only one Audio Listener anyways. The ECS way to do this is to:
- Create a new AudioListener component (somehow enforce only one being created, perhaps a singleton)
- Query for it from within the AudioManager's update method and then set the audio listener location then.

We could also just tie audio listening to the main camera and maintain that constraint across all scenes. Given your velocity on this, I recommend that as the approach. In this case, all you have to do is change AudioManager::Update's body to:

```cpp

void AudioManager::Update(float deltaTime) {
    Entity* cameraEnt = EntityManager::getInstance().findFirstEntityByDisplayName("Main Camera");
    updateListenerPosition(cameraEnt);
    
    std::vector<UUIDv4::UUID> audioEntitiesUUIDs = EntityManager::getInstance().findEntitiesByComponent(ComponentType::Audio);
// rest as before
}
```

Now you can remove the call before update.

Reply:
Good idea, will make audioManager have a member containing the listener location

Reply:
AudioManager - which is named like an entity, is actually an ECS system according to the source code. Conceptually that makes it confusing. 

I disagree with the idea that you should make audiomanager have a member because it's supposed to represent an ECS system which operates on entities that contain data, and not store extraneous data itself.

My suggestion is that in the scene definition you attach the audiolistener component to the camera, and use the camera's entity position as the listener position. Jas gives you the source code above which is pretty much doing that.